pallets/liquidity-pools-forwarder/src/lib.rs

```diff
@@ -0,0 +1,242 @@
+			Origin = DomainAddress,
+			Message = Vec<u8>,
+		>;
+
+		/// The entity which acts on unwrapped messages.
+		type MessageReceiver: MessageReceiver<Middleware = Self::RouterId, Origin = DomainAddress>;
+
+		/// An identification of a router.
+		type RouterId: Parameter + MaxEncodedLen;
+
+		/// The type that provides the router available for a domain.
+		type RouterProvider: RouterProvider<Domain, RouterId = Self::RouterId>;
+
+		// type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub (super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Forwarding info was set
+		ForwarderSet {
+			router_id: T::RouterId,
+			source_domain: Domain,
+			forwarding_contract: H160,
+		},
+		/// Forwarding info was removed
+		ForwarderRemoved {
+			router_id: T::RouterId,
+			source_domain: Domain,
+			forwarding_contract: H160,
+		},
+	}
+
+	/// Maps a router id to its forwarding info.
+	///
+	/// Can only be mutated via admin origin.
+	#[pallet::storage]
+	pub type RouterForwarding<T: Config> =
+		StorageMap<_, Blake2_128Concat, T::RouterId, ForwardInfo, OptionQuery>;
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The router id does not have any forwarder info stored
+		ForwardInfoNotFound,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Set forwarding info for the given router id.
+		///
+		/// Origin: Admin.
+		///
+		/// NOTE: Simple weight due to origin requirement.
+		#[pallet::weight(T::DbWeight::get().writes(1))]
+		#[pallet::call_index(0)]
+		pub fn set_forwarder(
+			origin: OriginFor<T>,
+			router_id: T::RouterId,
+			source_domain: Domain,
+			forwarding_contract: H160,
+		) -> DispatchResult {
+			T::AdminOrigin::ensure_origin(origin)?;
+
+			RouterForwarding::<T>::insert(
+				&router_id,
+				ForwardInfo {
+					source_domain,
+					contract: forwarding_contract,
+				},
+			);
+
+			Self::deposit_event(Event::<T>::ForwarderSet {
+				router_id,
+				source_domain,
+				forwarding_contract,
+			});
+
+			Ok(())
+		}
+
+		/// Remove the forwarding info for the given router id.
+		///
+		/// Origin: Admin.
+		///
+		/// NOTE: Simple weight due to origin requirement.
+		#[pallet::weight(T::DbWeight::get().writes(1))]
+		#[pallet::call_index(1)]
+		pub fn remove_forwarder(origin: OriginFor<T>, router_id: T::RouterId) -> DispatchResult {
+			T::AdminOrigin::ensure_origin(origin)?;
+
+			if let Some(info) = RouterForwarding::<T>::take(&router_id) {
+				Self::deposit_event(Event::<T>::ForwarderRemoved {
+					router_id,
+					source_domain: info.source_domain,
+					forwarding_contract: info.contract,
+				});
+			}
+
+			Ok(())
+		}
+	}
+
+	impl<T: Config> MessageSender for Pallet<T> {
+		type Message = T::Message;
+		type Middleware = T::RouterId;
+		type Origin = DomainAddress;
+
+		fn send(
+			router_id: T::RouterId,
+			origin: DomainAddress,
+			message: T::Message,
+		) -> DispatchResult {
+			let payload = if let Some(info) = RouterForwarding::<T>::get(&router_id) {
```

Comment:
Nit - use match?

Reply:
To the best of my knowledge, if the check is binary, `if else` is preferred - which is the case here.

Reply:
Improved in [b39fa47](https://github.com/centrifuge/centrifuge-chain/pull/1971/commits/b39fa47187b959c5c8dd3d2c7c73f1b5b55b0563)